src/main/java/org/cryptomator/cryptofs/CryptoFileSystem.java

```diff
@@ -29,6 +30,15 @@ public abstract class CryptoFileSystem extends FileSystem {
 	 */
 	public abstract Path getPathToVault();
 
+	/**
+	 * Provides the {@link Path} to the (data) ciphertext from a given cleartext path.
+	 *
+	 * @param cleartextPath absolute path to the cleartext file or folder belonging to this {@link CryptoFileSystem}. Internally the path must be an instance of {@link CryptoPath}
+	 * @return the {@link Path} to ciphertext file or folder containing teh actual encrypted data
+	 * @throws IOException
+	 */
+	public abstract Path getPathToDataCiphertext(Path cleartextPath) throws IOException;
+
```

Comment:
1. let's call this method `getCiphertextPath` (if it wasn't for directories, I would even call it `getC9rPath`)
2. document known specific exceptions, such as `ProviderMismatchException`
3. add description to `@throws IOException`

Reply:
Resolved in https://github.com/cryptomator/cryptofs/pull/154/commits/8167d5d6dc60b36d1f7ee6baefb489c7cbdccf8e